webapp/components/search_bar.jsx

```diff
@@ -171,7 +171,16 @@ export default class SearchBar extends React.Component {
 
     handleSubmit(e) {
         e.preventDefault();
-        this.handleSearch(this.state.searchTerm.trim());
+        var terms = this.state.searchTerm.trim();
+        
+        AppDispatcher.handleServerAction({
+            type: ActionTypes.RECEIVED_SEARCH_TERM,
+            term: terms,
+            do_search: true,
+            is_mention_search: false
+        });
```

Comment:
Fires an event when a new search is submitted, so that listeners know that search is in progress. This was already firing when the flagged/mentions buttons were clicked, but not when a new search was sent